### PR TITLE
[consensus] Dedicated channel for proposal buffering and batch process

### PR DESCRIPTION
**Pull request by zekun000**

### CODE REVIEW

File: consensus/src/round_manager_test.rs [hunk 1]
```diff
@@ -1683,11 +1683,11 @@ pub fn forking_retrieval_test() {
 
         println!("Process all local timeouts");
         for node in nodes.iter_mut() {
-            info!("Timeouts on {}", node.id);
+            println!("Timeouts on {}", node.id);
             for i in 0..timeout_votes {
-                info!("Timeout {} on {}", i, node.id);
+                println!("Timeout {} on {}", i, node.id);
```
Comment (sitalkedia) [inline consensus/src/round_manager_test.rs:1688]: nit: println -> info

**Review by sitalkedia — APPROVED** [review]
Did you observe improvement to remote fetch after this change in Forge?

**Comment by sitalkedia** [comment]
Hmm the change still doesn't help with reducing the remote block fetch - https://aptoslabs.grafana.net/goto/Lo8fHRWSg?orgId=1 compare without this change - https://aptoslabs.grafana.net/d/consensus/consensus?from=1695346868000&to=1695347743000&var-Datasource=fHo-R604z&var-BigQuery=axNEitxVz&var-metrics_source=All&var-chain_name=forge-1&var-cluster=All&var-namespace=forge-e2e-pr-10180&var-kubernetes_pod_name=All&orgId=1&viewPanel=25

So there must be something else we are missing.

File: consensus/src/round_manager.rs [hunk 2]
```diff
@@ -957,26 +946,61 @@ impl RoundManager {
...
-            futures::select! {
-                (peer_id, event) = event_rx.select_next_some() => {
-                    let result = match event {
-                        VerifiedEvent::ProposalMsg(proposal_msg) => {
-                            monitor!(
-                                "process_proposal",
-                                self.process_proposal_msg(*proposal_msg).await
-                            )
+            tokio::select! {
+                biased;
```
Comment (sitalkedia) [inline consensus/src/round_manager.rs:956]: Are you sure this works or should we be using the `select_biased` macro instead?

Comment (zekun000) [reply consensus/src/round_manager.rs:956]: according to this https://michael-f-bryan.github.io/linkcheck/tokio/macro.select.html it should work

**Comment by sitalkedia** [comment]
> Hmm the change still doesn't help with reducing the remote block fetch - https://aptoslabs.grafana.net/goto/Lo8fHRWSg?orgId=1 compare without this change - https://aptoslabs.grafana.net/d/consensus/consensus?from=1695346868000&to=1695347743000&var-Datasource=fHo-R604z&var-BigQuery=axNEitxVz&var-metrics_source=All&var-chain_name=forge-1&var-cluster=All&var-namespace=forge-e2e-pr-10180&var-kubernetes_pod_name=All&orgId=1&viewPanel=25
> 
> So there must be something else we are missing.

Any idea why this change doesn't help much with reducing the remote fetch?

**Comment by zekun000** [comment]
> Any idea why this change doesn't help much with reducing the remote fetch?

i don't, probably need to look into the logs

**Review by msmouse — APPROVED** [review]

**Review by msmouse — APPROVED** [review]

**Pull request merged**